dv-export/python/query-changes/dv_manager.py

```diff
@@ -2,109 +2,127 @@
 import logging
 import os
 import time
-from typing import Any, Tuple
+from typing import Tuple, List, Optional
 
 import pandas as pd
-from visier.api import DVExportApiClient
-from visier.connector import VisierSession
-
-from constants import *
+from visier_api_data_out import DataVersionExportApi, DataVersionExportDataVersionsDTO, \
+    DataVersionExportScheduleJobRequestDTO, DataVersionExportTableDTO, DataVersionExportDTO
 
 logger = logging.getLogger(__name__)
 
 
 class DVManager:
     """ Retrieves DVs, executes DV export jobs, and reads changes. """
 
-    def __init__(self, session: VisierSession,
+    def __init__(self,
+                 dv_api: DataVersionExportApi,
                  save_export_files_on_disk: bool,
                  export_files_path: str,
                  job_status_poll_interval_sec: int = 30,
                  job_timeout_sec: int = 1200):
-        self.dv_client = DVExportApiClient(session, raise_on_error=True)
+        self.dv_api = dv_api
         self.save_export_files_on_disk = save_export_files_on_disk
         self.export_files_path = export_files_path
         self.job_status_poll_interval_sec = job_status_poll_interval_sec
         self.job_timeout_sec = job_timeout_sec
 
-    def get_data_versions(self):
+    def get_data_versions(self) -> DataVersionExportDataVersionsDTO:
         """Retrieve a list of DVs available to export."""
-        return self.dv_client.get_data_versions_available_for_export().json()[DATA_VERSIONS]
+
+        return self.dv_api.get_available_data_versions()
 
     def get_export_data_version_times(self, export_uuid: str) -> Tuple[int, int]:
         """Retrieve a list of the data versions available to export."""
-        export_metadata = self.dv_client.get_data_version_export_metadata(export_uuid).json()
-        data_versions = self.get_data_versions()
-
-        base_data_version = next(
-            (dv for dv in data_versions if dv[DATA_VERSION] == export_metadata[BASE_DATA_VERSION_NUMBER]))
-        data_version = next(
-            (dv for dv in data_versions if dv[DATA_VERSION] == export_metadata[DATA_VERSION_NUMBER]))
-        base_dv_time = int(base_data_version[CREATED])
-        dv_time = int(data_version[CREATED])
+
+        export_metadata = self.dv_api.get_export(export_uuid)
+        if (export_metadata is None
```

Comment:
Should we error out if there > 1 exports returned?

Reply:
Encapsulated the method in _get_export_data_version. It returns only one DataVersionExportDTO.